instrumentation/jdbc/library/src/main/java/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriver.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Includes work from:
+/*
+ * Copyright 2017-2021 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.opentelemetry.instrumentation.jdbc;
+
+import io.opentelemetry.instrumentation.jdbc.internal.DbInfo;
+import io.opentelemetry.instrumentation.jdbc.internal.JdbcConnectionUrlParser;
+import io.opentelemetry.instrumentation.jdbc.internal.JdbcMaps;
+import io.opentelemetry.instrumentation.jdbc.internal.OpenTelemetryConnection;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.DriverPropertyInfo;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Properties;
+import java.util.logging.Logger;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+public class OpenTelemetryDriver implements Driver {
+
+  private static final String INTERCEPTOR_MODE_URL_PREFIX = "jdbc:otel:";
+
+  private static final OpenTelemetryDriver INSTANCE = new OpenTelemetryDriver();
+
+  private static boolean registered = false;
+  private static boolean interceptorMode = false;
+
+  static {
+    try {
+      register();
+    } catch (SQLException e) {
+      throw new ExceptionInInitializerError(e);
+    }
+  }
+
+  /**
+   * Ensure {@code TracingDriver} be the first driver of {@link DriverManager} to make sure
+   * "interceptor mode" works. WARNING: Driver like Oracle JDBC may fail since it's destroyed
+   * forever after deregistration.
+   */
+  public static synchronized void ensureRegisteredAsTheFirstDriver() {
+    try {
+      Enumeration<Driver> enumeration = DriverManager.getDrivers();
+      List<Driver> drivers = new ArrayList<>();
+      for (int i = 0; enumeration.hasMoreElements(); ++i) {
+        Driver driver = enumeration.nextElement();
+        if (i == 0 && driver == INSTANCE) {
+          // the first driver is OTEL driver, skip all this verification
+          return;
+        }
+        if (driver instanceof OpenTelemetryDriver) {
```

Comment:
Shouldn't it be the opposite?
```suggestion
        if (!(driver instanceof OpenTelemetryDriver)) {
```

Reply:
Yeah, fixed.